R/boxr_collab.R

```diff
@@ -151,6 +156,98 @@ box_invite <- function(item, accessible_by, role, can_view_path = FALSE) {
       .sep = " "
     )
   )
+  invisible(resp)
+}
+
+#' @rdname box_create_collab
+#' @keywords deprecated
+#' @export
+box_dir_invite <- function(dir_id, user_id, login = NULL, role = "viewer", 
+                           can_view_path = FALSE) {
+  
+  # if login is provided, ignore user_id
+  if (!is_void(login)) {
+    user_id <- NULL
+  }
+  
+  item <-
+    list(
+      type = "folder", # imagine box_file_invite()
+      id = as.character(dir_id)
+    )
+  
+  accessible_by <-
+    list(
+      type = "user", #  imagine inviting a group
+      id = as.character(user_id),
+      login = login
+    )
+  
+  box_create_collab_internal(item, accessible_by, role, can_view_path)
+} 
+
+#' Get existing collaborations
+#' 
+#' You must specify either `dir_id` or `file_id`, if both a specified `dir_id` is used.
+#' 
+#' @inheritParams box_dl
+#' @inheritParams box_fetch
+#' 
+#' @return Invisible `data.frame` with one row per collaboration.
+#' 
+#' @export
+box_get_collab <- function(dir_id = NULL, file_id = NULL) {
+  # detect item type for API call
+  item_id <- dir_id %|0|% file_id
```

Comment:
maybe you could use a helper function that could also be used with the create function, something like:

```r
get_item <- function(dir_id = NULL, file_id = NULL) {

  # assert that you have exactly one

  # logic to get id and type

  list(id = item_id, type = item_type)
}
```